synapse/axon.py

```diff
@@ -286,6 +345,25 @@ async def __anit__(self, dirn, conf=None):  # type: ignore
 
         self._initAxonHttpApi()
 
+    @contextlib.asynccontextmanager
+    async def holdHashLock(self, hashbyts):
+        '''
+        A context manager that synchronizes edit access to a blob.
+        '''
+
+        item = self.hashlocks.get(hashbyts)
+        if item is None:
+            self.hashlocks[hashbyts] = item = [0, asyncio.Lock()]
```

Comment:
Since this lock around adding a given hash protects that individual hash's addition (or removal), do we need fini behavior to be aware of this?

Reply:
resolving per discussion